libvast/src/expression_visitors.cpp

```diff
@@ -35,6 +35,52 @@
 
 namespace vast {
 
+expression expander::operator()(caf::none_t) const {
+  return expression{};
+}
+
+expression expander::operator()(const conjunction& c) const {
+  conjunction result;
+  for (auto& op : c)
+    result.push_back(caf::visit(*this, op));
+  return result;
+}
+
+expression expander::operator()(const disjunction& d) const {
+  disjunction result;
+  for (auto& op : d)
+    result.push_back(caf::visit(*this, op));
+  return result;
+}
+expression expander::operator()(const negation& n) const {
+  return {negation{caf::visit(*this, n.expr())}};
+}
+
+expression expander::operator()(const predicate& p) const {
+  auto make_addr_pred
```

Comment:
This function urgently needs comments above every step. Make it easy for the future maintainer to follow this function. `make_addr_pred` as a function name alone is not sufficient here.